app/uk/gov/hmrc/gform/gformbackend/GformConnector.scala

```diff
@@ -119,12 +119,31 @@ class GformConnector(ws: WSHttp, baseUrl: String) {
             }
 
           }
-          .recover {
+          .recoverWith {
             case UpstreamErrorResponse.WithStatusCode(statusCode, _) if statusCode == StatusCodes.NotFound.intValue =>
               logger.info(
                 s"Attempt to access form $formIdData, but form not found in MongoDB, attempt to look for $formIdDataOriginal as a fallback failed."
               )
-              None
+              def getFirstTemplateByLegacyIds(legacyIds: List[FormTemplateId]): Future[Option[FormTemplate]] =
+                legacyIds match {
+                  case Nil => Future.successful(None)
+                  case legacyFormId :: tail =>
+                    getFormTemplate(legacyFormId).map(Some(_)).recoverWith {
+                      case UpstreamErrorResponse.WithStatusCode(statusCode, _)
+                          if statusCode == StatusCodes.NotFound.intValue =>
+                        getFirstTemplateByLegacyIds(tail)
+                    }
+                }
```

Comment:
We shouldn't do any assumption about existence of `FormTemplate`. Or rather we should assume it does not exist, but user data can still be linked to the nonexistent `FormTemplate`. 

In other word need to check all `formTemplate.legacyFormIds` for existence of user data without making any assumption about corresponding `FormTemplate`.